test/batch_services.jl

```diff
@@ -101,6 +101,25 @@ unwrap(e::CapturedException) = unwrap(e.ex)
         pool = multibatcher.workers
         @test length(pool) == count_ready_workers(pool) == length(worker_ids) - 1
 
+        @testset "local manager doesn't mutate init state" begin
+            b1 = Batcher(myid(), worker_ids[2:2], batches; start=false)
+            b2 = Batcher(worker_ids[1], worker_ids[2:2], batches; start=false)
+
+            init_state = StableRNG(1)
+
+            start!(b1, init_state)
+            start!(b2, init_state)
+
+            x1, s1 = take!(b1, init_state)
+            x2, s2 = take!(b2, init_state)
+
+            @test x1 == x2
+            @test s1 == s2 != init_state
+
+            # not mutated
+            @test init_state == StableRNG(1)
+        end
```

Comment:
do we need to `stop!` the batcher or anything?

Reply:
whoops, yeah maybe.  shouldn't be a huge issue though these batches are small by design...

Reply:
what happens if we don't? is there just a bunch of unallocated memory hanging around?

Reply:
yeah they get cleaned up when the suite finishes.  dangling tasks are generally frowned upon but in tests it's not the end of the world...